src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.CompilationTrackerState.cs

```diff
@@ -115,35 +115,12 @@ private abstract class CompilationTrackerState
                         driver: null,
                         documentsAreFinal: false));
 
-                /// <summary>
-                /// The best compilation that is available that source generators have not ran on. May be an
-                /// in-progress, full declaration,  a final compilation, or <see langword="null"/>.
-                /// </summary>
-                public Compilation? CompilationWithoutGeneratedDocuments { get; }
```

Comment:
moved this to the new `WithCompilationState` type, where it is non-null.  So now callers just need to distinguish between between `NoCompilationState` and `WithCompilationState` types